internal/watcher/watcher.go

```diff
@@ -174,52 +182,156 @@ func (d Differential) WatchFn(path string) error {
 		return nil
 	}
 
-	// This is to allow for changes to propagate on the filesystem. If the file
-	// is large, the write won't be atomic. It will happen in, for example, 4096
-	// bytes chunks. 1 ms should be enough.
-	time.Sleep(time.Millisecond)
+	time.Sleep(10 * time.Millisecond)
 
-	contents, err := ioutil.ReadFile(path)
+	info, err := os.Stat(path)
 	if err != nil {
-		return fmt.Errorf("ioutil.ReadFile: %w", err)
+		return fmt.Errorf("error read file status %w", err)
 	}
 
-	checksum := md5.Sum(contents) //nolint:gosec
+	if info.Size() >= MaxFileSize {
```

Comment:
I think the code would benefit from creating a function that handles the case when `if info.Size() >= MaxFileSize `.  

Would it make sense to do something like the following and move the contents of the if into that separate function?

```go
if info.Size() >= MaxFileSize {
		d.WatchLargeFileFn(info, path)
		return nil
}
```